### DISCUSSION TITLE
Saving a model specification

### DISCUSSION THREAD

**Discussion by dmbates** [JuliaStats/MixedModels.jl#583] [Ideas]
I have been fitting models to the EnglishLexicon LDT (Lexical Decision Task) rate data.  Some of these models take several hours to fit on our cluster.  To quote these models in a publication it would be helpful to provide a complete model specification which would include a description of how to reproduce the data set, the model formula, the constrasts specification, and the optsum.  We already have saveoptsum/restoreoptsum using JSON and the data can be saved as one or more Arrow files.

What would be good ways of saving the model formula and the contrasts specification? (@palday, @kleinschmidt)

**Comment by palday** [comment]
Naively for the formula, we can just use `string(formula(model))`, but since the formula already has a schema applied to it which encodes the contrasts, that seems less than ideal.
